src/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1,27 +1,3041 @@
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_inlet_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_outlet_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            precooler_supply_pipe: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(1.),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            engine_starter_container: PneumaticPipe::new(
+                Volume::new::<cubic_meter>(0.5),
+                Pressure::new::<psi>(14.7),
+                ThermodynamicTemperature::new::<degree_celsius>(15.),
+            ),
+            engine_starter_exhaust: PneumaticExhaust::new(1e-2),
+            engine_starter_valve: DefaultValve::new_closed(),
+            precooler: Precooler::new(5.),
+        }
+    }
+
+    fn update(
+        &mut self,
+        context: &UpdateContext,
+        high_pressure_valve_controller: &impl ControllerSignal<HighPressureValveSignal>,
+        pressure_regulating_valve_controller: &impl ControllerSignal<PressureRegulatingValveSignal>,
+        engine_starter_valve_controller: &impl ControllerSignal<EngineStarterValveSignal>,
+        fan_air_valve_controller: &impl ControllerSignal<FanAirValveSignal>,
+        engine: &(impl EngineCorrectedN1 + EngineCorrectedN2),
+    ) {
+        // Update engines
+        self.fan_compression_chamber_controller
+            .update(context, engine);
+        self.intermediate_pressure_compression_chamber_controller
+            .update(context, engine);
+        self.high_pressure_compression_chamber_controller
+            .update(context, engine);
+
+        self.fan_compression_chamber
+            .update(&self.fan_compression_chamber_controller);
+        self.intermediate_pressure_compression_chamber
+            .update(&self.intermediate_pressure_compression_chamber_controller);
+        self.high_pressure_compression_chamber
+            .update(&self.high_pressure_compression_chamber_controller);
+
+        // Update valves (open amount)
+        self.high_pressure_valve
+            .update_open_amount(high_pressure_valve_controller);
+        self.pressure_regulating_valve
+            .update_open_amount(pressure_regulating_valve_controller);
+        self.engine_starter_valve
+            .update_open_amount(engine_starter_valve_controller);
+        self.fan_air_valve
+            .update_open_amount(fan_air_valve_controller);
+
+        // Update valves (fluid movement)
```

Comment:
**thought**: The function names are good enough for this and some of the other comments in this function to not really be necessary.
```suggestion
```